controllers/providers/aws/aws.go

```diff
@@ -507,3 +529,210 @@ func IsStackInConditionState(key string, condition string) bool {
+		PolicyArn: aws.String(defaultPolicyArn),
+		RoleName:  w.CreateDefaultRoleName(),
+	}
+	_, err := w.IamClient.DetachRolePolicy(rolePolicy)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			if aerr.Code() == iam.ErrCodeNoSuchEntityException {
+				return false, nil
+			}
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+func (w *AwsFargateWorker) DeleteDefaultRole() (bool, error) {
+	role := &iam.DeleteRoleInput{
+		RoleName: w.CreateDefaultRoleName(),
+	}
+	_, err := w.IamClient.DeleteRole(role)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			if aerr.Code() == iam.ErrCodeNoSuchEntityException {
+				return false, nil
+			}
+		}
+		return false, err
+	}
+	return true, nil
+}
+func (w *AwsFargateWorker) CreateDefaultRoleName() *string {
+	s := fmt.Sprintf("%v_%v_Role", *w.ClusterName, *w.ProfileName)
+	return &s
+}
+
+func (w *AwsFargateWorker) GetDefaultRole() (*iam.Role, error) {
+	var roleName = w.CreateDefaultRoleName()
+	role := &iam.GetRoleInput{
+		RoleName: roleName,
+	}
+	resp, err := w.IamClient.GetRole(role)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.Role, nil
+}
+func (w *AwsFargateWorker) CreateDefaultRole() (bool, error) {
+	var roleName = w.CreateDefaultRoleName()
+	var template = `{"Version":"2012-10-17","Statement":[{"Effect":"Allow","Principal":{"Service":"eks-fargate-pods.amazonaws.com"},"Action":"sts:AssumeRole"}]}`
+	role := &iam.CreateRoleInput{
+		AssumeRolePolicyDocument: &template,
+		Path:                     aws.String("/"),
+		RoleName:                 roleName,
+	}
+	_, err := w.IamClient.CreateRole(role)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			if aerr.Code() == iam.ErrCodeEntityAlreadyExistsException {
+				return false, nil
+			}
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+func (w *AwsFargateWorker) AttachDefaultPolicyToDefaultRole() error {
+	rolePolicy := &iam.AttachRolePolicyInput{
+		PolicyArn: aws.String(defaultPolicyArn),
+		RoleName:  w.CreateDefaultRoleName(),
+	}
+	_, err := w.IamClient.AttachRolePolicy(rolePolicy)
+	return err
+}
+
+func (w *AwsFargateWorker) CreateProfile(arn *string) error {
+	input := &eks.CreateFargateProfileInput{
+		ClusterName:         w.ClusterName,
+		FargateProfileName:  w.ProfileName,
+		PodExecutionRoleArn: arn,
+		Selectors:           w.Selectors,
+		Subnets:             w.Subnets,
+		Tags:                w.Tags,
+	}
+
+	_, err := w.EksClient.CreateFargateProfile(input)
+	return err
+}
+
+func (w *AwsFargateWorker) DeleteProfile() error {
+	input := &eks.DeleteFargateProfileInput{
+		ClusterName:        w.ClusterName,
+		FargateProfileName: w.ProfileName,
+	}
+	_, err := w.EksClient.DeleteFargateProfile(input)
+	return err
+}
+
+func (w *AwsFargateWorker) DescribeAllProfiles(profiles []*string) ([]*eks.FargateProfile, error) {
+	fargateProfiles := []*eks.FargateProfile{}
+	for _, profile := range profiles {
+		x, err := DescribeProfileWithParms(w.EksClient, w.ClusterName, profile)
```

Comment:
This can get expensive from api limits perspective (imagine an account with 50 profiles that keeps reconciling them) - definitely OK for now, but we should explore caching down the road for these kind of calls.
I've created #87 to address this.

Reply:
Another option is to NOT check this at all - if you only need this for the case of creation (can't create when other profile is deleting), have a requeue logic in case where the create fails due to another profile being deleted (or have a hard failure for very specific error codes).
In that case the create will fail and keep retrying (via requeue) - this way we have less code, and make less API calls.

Reply:
When I tried creating a profile while one was being deleted, it left the profile in some partial state. I had to delete the profile first.  So I could definitely create reconciliation logic to delete the profile when it ended up in this partial state.  I could save some state so I would know to attempt another create after deleting the profile.  WDYT?

Reply:
I pulled out all the logic associated with CanDelete().  It had a race condition anyway.  I now detect the eks.ErrorCodeResourceInUse on the profile creation and deletion and rely upon the reconciliation to retry.